doc/release-notes-25355.md

```diff
@@ -0,0 +1,14 @@
+Notable changes
+===============
+
+P2P and network changes
+-----------------------
+
+- With I2P connections, a new, transient address is used for each outbound
+  connection by default. (#25355)
+
+New settings
+------------
+
+- The `-i2ptransientout` option has been added to generate a new I2P address
```

Comment:
```suggestion
- An `-i2ptransientout` configuration option has been added to opt out of using a new, transient I2P address
```

Reply:
I find the suggestion unnecessary long. Maybe append "(default: true)" to the current text?

Reply:
Could drop "of the new default behavior" for brevity if you prefer, but opting out of the new default is the only reason for adding the config option so it seems best to describe it in this way.

Reply:
If there is an option to opt out of X, I would expect enabling that option to opt out of X. This is not the case with `-i2ptransientout`, thus I think the above suggestion could be misleading. That option is to enable the usage of transient addresses. To opt out of that the user has to disable the option.

Reply:
Closing as not relevant anymore - `-i2ptransientout` was dropped.